support/cas-server-support-trusted-mfa/src/main/java/org/apereo/cas/trusted/web/flow/DeviceFingerprintComponent.java

```diff
@@ -0,0 +1,42 @@
+package org.apereo.cas.trusted.web.flow;
+
+import org.springframework.core.Ordered;
+import org.springframework.webflow.execution.RequestContext;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+/**
+ * Interface for determining a device fingerprint component for usage within MFA trusted device records.
+ *
+ * @author Daniel Frett
+ * @since 5.3.0
+ */
+@FunctionalInterface
+public interface DeviceFingerprintComponent extends Ordered {
```

Comment:
The suffix "Component" seems a bit redundant to me; Everything is a component. What exactly the purpose of the interface? That should likely be indicated in the name.

Reply:
hmm, I used `Component` in the sense that the object implementing this interface generates a component that gets combined together in `DeviceFingerprintStrategy` to produce a full `deviceFingerprint`.

maybe `DeviceFingerprintDimension` instead?

Reply:
I think you should opt for a verb that encompasses the category of features this interface provides rather than a noun since the sort of class we have here actually does stuff, rather carrying stuff under a label. Its sole job seems to deal with calculating a fingerprint *hash* for the device. So anything in the neighborhood of "Calculator" seems OK with me; "Computer", "Resolver", etc. 

The thing that it returns however, after doing the calculation were it not a simple String could perhaps be refactored into a `DeviceFingerprintDimension` which one might see as a simple data class, carrying data only. 

Please note that I am possibly nitpicking here. If `DeviceFingerprintComponent` is a good enough, so be it :)

Reply:
I'm going to let this rest for the night and re-examine it tomorrow with a fresh mind :)

Reply:
also, good point on the verb vs noun. That actually makes it easier to come up with a better name :)